src/components/forms/ChooseTemplate/ChooseTemplate.tsx

```diff
@@ -3,12 +3,22 @@
 // we need this to be a client component because nhsuk-react-components uses client-only react features
 
 import { FC } from 'react';
-import { PageComponentProps } from '../../../utils/types';
-import { NHSNotifyRadioButtonForm } from '../../molecules/NHSNotifyRadioButtonForm/NHSNotifyRadioButtonForm';
-import { ZodErrorSummary } from '../../molecules/ZodErrorSummary/ZodErrorSummary';
-import { chooseTemplatePageContent } from '../../../content/content';
+import { useFormState } from 'react-dom';
+import { chooseTemplateAction } from '@forms/ChooseTemplate/server-action';
+import { PageComponentProps } from '@utils/types';
+import { NHSNotifyRadioButtonForm } from '@molecules/NHSNotifyRadioButtonForm/NHSNotifyRadioButtonForm';
+import { ZodErrorSummary } from '@molecules/ZodErrorSummary/ZodErrorSummary';
+import { chooseTemplatePageContent } from '@content/content';
+import { useRouter } from 'next/navigation';
+
+export const ChooseTemplate: FC<PageComponentProps> = ({ initialState }) => {
+  const [state, action] = useFormState(chooseTemplateAction, initialState);
+  const router = useRouter();
+
+  if (state.redirect) {
+    router.push(state.redirect);
```

Comment:
using router.push like this appears to be much faster than using the nextjs redirect function